### DISCUSSION TITLE
Support for resp3

### DISCUSSION THREAD

**Discussion by monkey92t** [redis/go-redis#1728] [General]
Support for redis.RESP3, it is difficult to maintain the current performance of RESP2. According to my test, it will cause a 5-10% performance drop, which is about an increase of 100-400ns/op(The way to use #1716).
This is because the RESP3 protocol adds an Attribute type, which looks like this:

```go
*3<CR><LF>
    :1<CR><LF>
    :2<CR><LF>
    |1<CR><LF>
        +ttl
        :3600
    :3<CR><LF>

> r = Redis.new
#<Redis client>

> r.mget("a","b", "c")
#<Redis reply>

> r
[1,2,3]

> r[2].attribs
{"ttl": 3600}
```

The current command->proto.Reader cannot handle it, and a processing method similar to #1716 (proto.Value) needs to be used, unless we abandon this type.

**Comment by monkey92t** [comment]
@vmihailenco What's your opinion?

**Reply by vmihailenco** [reply]
For the sake of others the spec is at https://github.com/antirez/RESP3/blob/master/spec.md#attribute-type

I believe we should just ignore the attribute resp type for now (read and discard it). If/when users need to access the attributes we can provide a separate API that will return the `proto.Value` you've mentioned:

```go
rdb.MGet(ctx, "a", "b") // continues to work as usually

vals, err := rdb.Resp3(ctx, "mget", "a", "b").Result()
print(vals) // []proto.Value{proto.Value{}, proto.Value{}}
```

Does that make sense? Are there any other issues with implementing RESP3?
